Unix/base/batch.h

```diff
@@ -113,68 +113,138 @@ void Batch_InitFromBuffer(
 void Batch_Destroy(
     Batch* self);
 
-Batch* Batch_New(
-    size_t maxPages);
-
 void Batch_Delete(
     Batch* self);
 
-void* Batch_Get(
+#define Batch_New(pages) __Batch_New(BATCH_CALLSITE pages)
+#define Batch_Get(self, size) __Batch_Get(BATCH_CALLSITE self, size)
+#define Batch_GetClear(self, size) __Batch_GetClear(BATCH_CALLSITE self, size)
+#define Batch_Tcsdup(self, str) __Batch_Tcsdup(BATCH_CALLSITE self, str)
+#define Batch_Strdup(self, str) __Batch_Strdup(BATCH_CALLSITE self, str)
+#define Batch_StrTcsdup(self, str) __Batch_StrTcsdup(BATCH_CALLSITE self, str)
+#define Batch_ZStrdup(self, str) __Batch_ZStrdup(BATCH_CALLSITE self, str)
+
+#if defined(CONFIG_ENABLE_DEBUG) && defined(CONFIG_OS_LINUX)
+
+#define BATCH_CALLSITE __FILE__, __LINE__, __FUNCTION__,
```

Comment:
if you had another define for lets say:
#define BATCH_CALLSITE_DEF const char * file, size_t line, const char *func,
and
#define BATCH_CALLSITE_PASSTHROUGH file, line, func, 
then I _think_ it would be more or less the same code throughout except the macros for debug get filled out and are empty when not.
A similar thing was done for strand debug code throughout.